test/e2e/framework.go

```diff
@@ -529,21 +534,20 @@ func (data *TestData) deployAntreaIPSec() error {
 // deployAntreaFlowExporter deploys Antrea with flow exporter config params enabled.
 func (data *TestData) deployAntreaFlowExporter(ipfixCollector string) error {
 	// Enable flow exporter feature and add related config params to antrea agent configmap.
-	return data.mutateAntreaConfigMap(func(data map[string]string) {
-		antreaAgentConf, _ := data["antrea-agent.conf"]
-		antreaAgentConf = strings.Replace(antreaAgentConf, "#  FlowExporter: false", "  FlowExporter: true", 1)
-		if ipfixCollector != "" {
-			antreaAgentConf = strings.Replace(antreaAgentConf, "#flowCollectorAddr: \"flow-aggregator.flow-aggregator.svc:4739:tcp\"", fmt.Sprintf("flowCollectorAddr: \"%s\"", ipfixCollector), 1)
-		}
-		antreaAgentConf = strings.Replace(antreaAgentConf, "#flowPollInterval: \"5s\"", "flowPollInterval: \"1s\"", 1)
-		antreaAgentConf = strings.Replace(antreaAgentConf, "#flowExportFrequency: 12", "flowExportFrequency: 2", 1)
-		if testOptions.providerName == "kind" {
-			// In Kind cluster, there are issues with DNS name resolution on worker nodes.
-			// We will skip TLS testing for Kind cluster because the server certificate is generated with Flow aggregator's DNS name
-			antreaAgentConf = strings.Replace(antreaAgentConf, "#enableTLSToFlowAggregator: true", "enableTLSToFlowAggregator: false", 1)
-		}
-		data["antrea-agent.conf"] = antreaAgentConf
-	}, false, true)
+	ap := []configPair{
+		{"FlowExporter", "true"},
+		{"flowPollInterval", "\"1s\""},
+		{"flowExportFrequency", "2"},
+	}
+	if ipfixCollector != "" {
+		ap = append(ap, configPair{"flowCollectorAddr", fmt.Sprintf("flowCollectorAddr: \"%s\"", ipfixCollector)})
+	}
+	if testOptions.providerName == "kind" {
+		// In Kind cluster, there are issues with DNS name resolution on worker nodes.
+		// We will skip TLS testing for Kind cluster because the server certificate is generated with Flow aggregator's DNS name
+		ap = append(ap, configPair{"enableTLSToFlowAggregator", "false"})
+	}
+	return data.mutateAntreaConfigMap([]configPair{}, ap, false, true)
```

Comment:
nit: could use `nil` for the first argument

Reply:
Updated.